submissions/api.py

```diff
@@ -413,6 +413,63 @@ def get_all_submissions(course_id, item_id, item_type, read_replica=True):
         yield data
 
 
+def get_all_course_submission_information(course_id, item_type, read_replica=True):
+    """ For the given course, get all the submissions, student items, and scores of the given item type.
+
+    Args:
+        course_id (str): The course that we are getting submissions from.
+        item_type (str): The type of items that we are getting submissions for.
+        read_replica (bool): Try to use the database's read replica if it's available.
+
+    Yields:
+        A tuple of three dictionaries representing:
+        (1) a student item with the following fields:
+            student_id
+            course_id
+            student_item
+            item_type
+        (2) a submission with the following fields:
+            student_item
+            attempt_number
+            submitted_at
+            created_at
+            answer
+        (3) a score with the following fields:
+            student_item
+            submission
+            points_earned
+            points_possible
+            created_at
+            submission_uuid
+    """
+
+    submission_qs = Submission.objects
+    if read_replica:
+        submission_qs = _use_read_replica(submission_qs)
+
+    query = submission_qs.select_related('student_item').prefetch_related('score_set').filter(
+        student_item__course_id=course_id,
+        student_item__item_type=item_type,
+    ).iterator()
+
+    for submission in query:
+        student_item = submission.student_item
+        if submission.score_set.count() > 0:
+            for score in submission.score_set.all():
+                yield (
+                    StudentItemSerializer(student_item).data,
+                    SubmissionSerializer(submission).data,
+                    ScoreSerializer(score).data
```

Comment:
You could create `serialized_scores = []`, which would be `[{}]` in the case of no scores.  Then, loop over `serialized_scores` and call yield, without having to replicate the yield line.